### PR TITLE
[feat](stats) enable set auto analyze time by set global session variable

### PR DESCRIPTION
**Pull request by Kikyou1997**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Kikyou1997** [comment]
run buildall

**Review by zfr9527 — APPROVED** [review]
LGTM

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 1]
```diff
@@ -1144,6 +1150,14 @@ public void setMaxJoinNumberOfReorder(int maxJoinNumberOfReorder) {
     @VariableMgr.VarAttr(name = ENABLE_MEMTABLE_ON_SINK_NODE, needForward = true)
     public boolean enableMemtableOnSinkNode = false;
 
+    @VariableMgr.VarAttr(name = FULL_AUTO_ANALYZE_START_TIME, needForward = true)
+    public String fullAutoAnalyzeStartTime = "";
+
+    @VariableMgr.VarAttr(name = FULL_AUTO_ANALYZE_END_TIME, needForward = true)
+    public String fullAutoAnalyzeEndTime = "";
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:1157]: add chinese and english description

Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:1157]: add default value

Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:1157]: check format when set variable

Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:1157]: should use global variable, not session variable

**Review by morrySnow — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java (hunk 1 above)
Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java]: It's hard to implement over current framework

Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java]: If these variables aren't defined by user, we would use default config values in of FE

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 2]
```diff
@@ -1147,6 +1155,7 @@ public void setMaxJoinNumberOfReorder(int maxJoinNumberOfReorder) {
     @VariableMgr.VarAttr(name = ENABLE_MEMTABLE_ON_SINK_NODE, needForward = true)
     public boolean enableMemtableOnSinkNode = false;
 
+
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:1158]: remove this line

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 3]
```diff
@@ -397,6 +399,12 @@ public class SessionVariable implements Serializable, Writable {
 
     public static final String INVERTED_INDEX_CONJUNCTION_OPT_THRESHOLD = "inverted_index_conjunction_opt_threshold";
 
+    // For test only
+    public static final String FULL_AUTO_ANALYZE_START_TIME = "full_auto_analyze_start_time";
+
+    // For test only
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:405]: remove `full test only` comment

**Comment by Kikyou1997** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 4]
```diff
@@ -1159,6 +1165,18 @@ public void setMaxJoinNumberOfReorder(int maxJoinNumberOfReorder) {
                     + " use a skiplist to optimize the intersection."})
     public int invertedIndexConjunctionOptThreshold = 1000;
 
+    @VariableMgr.VarAttr(name = FULL_AUTO_ANALYZE_START_TIME, needForward = true, checker = "checkAnalyzeTimeFormat",
+            description = {"该参数定义自动ANALYZE例程的开始时间",
+                    "This parameter defines the start time for the automatic ANALYZE routine."},
+            flag = VariableMgr.GLOBAL)
+    public String fullAutoAnalyzeStartTime = "";
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:1172]: These should be in `GlobalVariable.java`

**Review by morningman — APPROVED** [review]
LGTM

**Review by morrySnow — APPROVED** [review]

**Pull request merged**